tsconfig.json

```diff
@@ -9,7 +9,7 @@
     "forceConsistentCasingInFileNames": true,
     "noEmit": true,
     "esModuleInterop": true,
-    "module": "esnext",
+    "module": "CommonJS",
```

Comment:
Whats this?

Reply:
This one is for running the typescript server file alone. I reverted this change and overridden the compiler options for this when 'ts-node' is the command

Reply:
Is there a need for running the server file alone tho

Reply:
What would you suggest? Another way would be to make a server class and move all the server logic there and create the server in one of the ts files

Reply:
Hmm, why can't we put the code in matchingServer in the matchingRouter?

Reply:
If the server code runs in the router directly, I believe it won't allow any other code to run in the router while the amqp server code is running, feel free to test it tho

Reply:
We could use a separate thread for this? https://threads.js.org/usage

Reply:
not sure if its worth tho. What is the use of matchingServer.ts? Haven't had time to fully understand it

Reply:
What it does is that it creates a channel with queues that can receive messages from client channels. This acts as our 'server' and we can send messages (matchRequests, cancelRequests etc) to it. Then, it performs matching every second and sends messages back to the client channels using their own respective queues, indicating that a match has been found. For now, the message is just the id of the other person matched with them and whether the request was successful or not. There is a timeout set to 30 seconds, where the request sent will be removed and the message sent to the respective client channel will indicate unsuccessful match. This is basically the gist of it.

For reference, I have included the page on RPC for rabbitmq, which will probably help a lot in visualizing it: 
https://www.rabbitmq.com/tutorials/tutorial-six-javascript.html

Reply:
Am I right if I say that we are sending the trpc requests and then the trpc server will send an rpc to amqp? Seems odd

Reply:
Yep, you can't run the amqp code on the browser (at least from what I can tell), so I chose to use the router to send the message to the amqp 'server'

https://stackoverflow.com/questions/45936495/uncaught-typeerror-webpack-require-connect-is-not-a-function-at-connec

Reply:
> If the server code runs in the router directly, I believe it won't allow any other code to run in the router while the amqp server code is running, feel free to test it tho. Plus, the server won't start until you call one of the router methods (when you click the button to queue)

Just tried to dump all the code in matchingServer into matchingRequest and ran next dev and it just work. Is there some specific workflow in which there is an error?

Reply:
Hm it didn't work earlier for me for some reason, I'll refactor the server file later then